xray/src/generation.rs

```diff
@@ -635,55 +626,23 @@ pub fn build_xray_quadtree(
     });
     progress_bar.lock().unwrap().finish_println("");
 
+    let mut all_nodes = FnvHashSet::default();
+    let mut previous_level_nodes = created_leaf_node_ids;
+    for node in &previous_level_nodes {
+        all_nodes.insert(*node);
+    }
     for current_level in (0..deepest_level).rev() {
-        let nodes_to_create: FnvHashSet<NodeId> = parents_to_create_rx.into_iter().collect();
-        let progress_bar = create_syncable_progress_bar(
-            nodes_to_create.len(),
-            &format!("Building level {}", current_level),
+        previous_level_nodes = build_level(
+            output_directory,
+            tile,
+            current_level,
+            &previous_level_nodes,
+            parameters,
         );
-        let (parents_to_create_tx, new_rx) = crossbeam::channel::unbounded();
-        parents_to_create_rx = new_rx;
-        nodes_to_create.into_par_iter().for_each(|node_id| {
-            all_nodes_tx.send(node_id).unwrap();
-
-            let mut children = [None, None, None, None];
-
-            // We a right handed coordinate system with the x-axis of world and images
-            // aligning. This means that the y-axis aligns too, but the origin of the image
-            // space must be at the bottom left. Since images have their origin at the top
-            // left, we need actually have to invert y and go from the bottom of the image.
-            for id in 0..4 {
-                let png = get_image_path(
-                    output_directory,
-                    node_id.get_child_id(&ChildIndex::from_u8(id)),
-                );
-                if !png.exists() {
-                    continue;
-                }
-                children[id as usize] = Some(image::open(&png).unwrap().to_rgba());
-            }
-            if children.iter().any(|child| child.is_some()) {
-                let large_image = build_parent(&children, parameters.tile_background_color);
-                let image = image::DynamicImage::ImageRgba8(large_image).resize(
-                    tile.size_px,
-                    tile.size_px,
-                    image::imageops::FilterType::Lanczos3,
-                );
-                image
-                    .as_rgba8()
-                    .unwrap()
-                    .save(&get_image_path(output_directory, node_id))
-                    .unwrap();
-                if let Some(id) = node_id.parent_id() {
-                    parents_to_create_tx.send(id).unwrap()
-                }
-            }
-            progress_bar.lock().unwrap().inc();
-        });
-        progress_bar.lock().unwrap().finish_println("");
-        drop(parents_to_create_tx);
+        for node in &previous_level_nodes {
+            all_nodes.insert(*node);
+        }
```

Comment:
How about shortening this to
```rust
all_nodes.extend(&previous_level_nodes);
```
?

Reply:
ah, thanks, this is exactly what i was looking for!